lib/ecosystems/enrich.go

```diff
@@ -198,9 +199,11 @@ func enrichComponentsWithEcosystems(bom *cdx.BOM, enrichFuncs []func(cdx.Compone
 	bom.Components = &newComponents
 }
 
-func EnrichSBOM(bom *cdx.BOM) *cdx.BOM {
+func EnrichSBOM(doc *sbom.SBOMDocument) *sbom.SBOMDocument {
```

Comment:
suggestion: since we are mutating the input, rather than returning a new instance, we might change the signature to 

```suggestion
func EnrichSBOM(doc *sbom.SBOMDocument) {
```